lib/makara/notifications.rb

```diff
@@ -0,0 +1,40 @@
+module Makara
+  # A notification handler that allows users to register arbitrary instrumentation callbacks. For instance, this can be
+  # used to fire dtrace or strace probes, or to write log messages. Callbacks are in the form of blocks, which can take
+  # 0 to N arguments, where N is the number of arguments provided to the the :notify!: method.
+  #
+  # Example:
+  #
+  #   Makara::Cache instruments the :write: method as follows:
+  #
+  #     Notifications.notify!('Cache:write', key, value, ttl)
```

Comment:
does your dtrace implementation have to know about all the keys. Think there is a use case for subscribing to `*`?